src/write.c

```diff
@@ -537,6 +550,35 @@ static avifBool avifEncoderDetectChanges(const avifEncoder * encoder, avifEncode
     return AVIF_TRUE;
 }
 
+// Same as 'avifEncoderWriteColorProperties' but for the colr nclx box only.
+static void avifEncoderWriteNclxProperty(avifRWStream * outputStream,
+                                         const avifImage * imageMetadata,
+                                         struct ipmaArray * ipma,
+                                         avifItemPropertyDedup * dedup)
+{
+    avifRWStream * s = outputStream;
+    if (dedup) {
+        assert(ipma);
+        // Use the dedup's temporary stream for box writes
+        s = &dedup->s;
+    }
```

Comment:
Move to arg, see https://github.com/AOMediaCodec/libavif/pull/1493#discussion_r1290782821 (also https://github.com/AOMediaCodec/libavif/pull/1493#discussion_r1291856361)

Same for `avifEncoderWriteHDRProperties()` below